pkg/transaction/jmsprotocol.go

```diff
@@ -0,0 +1,86 @@
+package transaction
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+// JMSProtocol - Represents the details in a transaction event for the JMS protocol
+type JMSProtocol struct {
+	Type             string `json:"type,omitempty"`
+	AuthSubjectID    string `json:"authSubjectId,omitempty"`
+	JMSMessageID     string `json:"jmsMessageID,omitempty"`
+	JMSCorrelationID string `json:"jmsCorrelationID,omitempty"`
+	JMSDestination   string `json:"jmsDestination,omitempty"`
+	JMSProviderURL   string `json:"jmsProviderURL,omitempty"`
+	JMSDeliveryMode  int    `json:"jmsDeliveryMode,omitempty"`
+	JMSPriority      int    `json:"jmsPriority,omitempty"`
+	JMSReplyTo       string `json:"jmsReplyTo,omitempty"`
+	JMSRedelivered   int    `json:"jmsRedelivered,omitempty"`
+	JMSTimestamp     int    `json:"jmsTimestamp,omitempty"`
+	JMSExpiration    int    `json:"jmsExpiration,omitempty"`
+	JMSType          string `json:"jmsType,omitempty"`
+	JMSStatus        string `json:"jmsStatus,omitempty"`
+	JMSStatusText    string `json:"jmsStatusText,omitempty"`
+}
+
+// ToMapStringString - convert the JMSProtocol to a map[string]string
+func (j *JMSProtocol) ToMapStringString() (map[string]string, error) {
+	jmsPropertyStringMap := make(map[string]string)
+
+	value := reflect.ValueOf(j).Elem()
+	for i := 0; i < value.NumField(); i++ {
+		field := value.Field(i)
+		jsonField := value.Type().Field(i).Tag.Get("json")
+		jsonKey := strings.Split(jsonField, ",")[0]
+
+		switch field.Kind() {
+		case reflect.String:
+			jmsPropertyStringMap[jsonKey] = field.String()
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+			jmsPropertyStringMap[jsonKey] = strconv.FormatInt(field.Int(), 10)
+		}
+	}
+	return jmsPropertyStringMap, nil
+}
+
+// FromMapStringString - convert the map[string]string to a JMSProtocol
+func (j *JMSProtocol) FromMapStringString(propertyMap map[string]string) error {
+	ignoreFields := map[string]bool{
+		"Type":         true,
+		"JMSStatus":    true,
+		"JMSTimestamp": true,
+	}
```

Comment:
Should I ignore these here (like I am), expect them to have been set on the object, or ignore these fields in the redaction?

@sbolosan

Reply:
yeah... To me, it looks ok to be here. Hmmm, yeah, i like it here